src/app/cli/src/init/client.ml

```diff
@@ -1764,11 +1764,9 @@ let generate_libp2p_keypair =
           match%bind
             Mina_net2.create ~logger ~conf_dir:tmpd ~all_peers_seen_metric:false
               ~pids:(Child_processes.Termination.create_pid_table ())
-              ~on_unexpected_termination:(fun () ->
```

Comment:
Ever call-site except for one just setup the behavior to raise an error and crash when the libp2p_helper encounters and unexpected error. The call-site that was actually doing something special on unexpected termination was already catching exceptions and handling them with similar logic. So I changed this functionality so that it always throws an exception if there is an unexpected termination of the libp2p_helper, and updated the catch logic in the one place that was actually utilizing this special argument to gracefully restart the helper.